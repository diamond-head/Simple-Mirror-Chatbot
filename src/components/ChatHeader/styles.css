.chat-header__container {
  padding: 18px;
  align-items: center;
  color: var(--white);
  border-radius: 0;
  box-shadow: 0 2px 7px 0 rgb(0 0 0 / 10%);
  display: flex;
  flex-shrink: 0;
  background-color: var(--primary-color);
  gap: 16px;
}


@media screen and (max-width: 414px) {
  .chat-header__container {
  }
}

.chat-header__icon img{
  height: 40px;
  width: 40px;
  border-radius: 50%;
}

.chat-header__title {
  font-size: 16px;
  font-weight: 600;
  line-height: 22px;
}