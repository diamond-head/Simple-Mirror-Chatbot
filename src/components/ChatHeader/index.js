import React from 'react';
import './styles.css';

export default function ChatHeader({ title, icon, ...props }) {
  return (
    <div className="chat-header__container">
      <div className="chat-header__icon">
        <img src={icon} alt='header-profile-icon'/>
      </div>
      <div className="chat-header__title">
        {title}
      </div>
    </div>
  );
}