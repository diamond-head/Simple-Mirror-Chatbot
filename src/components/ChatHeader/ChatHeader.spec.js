import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ChatHeader from './index';
import Icon from '../../assets/images/profile-icon.jpg'

describe("<ChatHeader /> component", () => {
  it("ChatHeader snapshot renders", () => {
    const component = renderer.create(
      <ChatHeader title="Amber" icon={Icon} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render an chat-header image", () => {
    const wrapper = shallow(<ChatHeader title="Amber" icon={Icon} />);
    expect(wrapper.find('img').prop('src')).toEqual(Icon)
  });

  it("renders an chat-header title", () => {
    const wrapper = shallow(<ChatHeader title="Amber" icon={Icon} />);
    expect(wrapper.find('.chat-header__title').text()).toBe("Amber");
  });
});
