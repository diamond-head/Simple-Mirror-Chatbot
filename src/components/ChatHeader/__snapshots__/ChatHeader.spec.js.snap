// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> component ChatHeader snapshot renders 1`] = `
<div
  className="chat-header__container"
>
  <div
    className="chat-header__icon"
  >
    <img
      alt="header-profile-icon"
      src="test-file-stub"
    />
  </div>
  <div
    className="chat-header__title"
  >
    Amber
  </div>
</div>
`;
